common/matrix/dense_kernels.hpp.inc

```diff
@@ -539,4 +539,48 @@ __global__ __launch_bounds__(default_block_size) void outplace_absolute_dense(
 }
 
 
+template <typename ValueType, typename ComplexType>
+__global__ __launch_bounds__(default_block_size) void make_complex(
+    size_type num_rows, size_type num_cols, const ValueType *__restrict__ in,
+    size_type stride_in, ComplexType *__restrict__ out, size_type stride_out)
```

Comment:
If this ComplexType is only for complex<ValueType>, I prefer use one template<typename ValueType> and use to_complex<ValueType> to represent complex<ValueType>. 
Before Ginkgo 2.0.0, I think we need to discuss the precision(ValueType) conversion...

Reply:
This doesn't work, since to_complex returns the CPU complex type, not thrust::complex. An alternative would be `thrust::complex<remove_complex<ValueType>>`

Reply:
you are right. maybe does another way `cuda_type<to_complex<ValueType>>` work?